prow/apis/prowjobs/v1/types.go

```diff
@@ -161,6 +161,18 @@ type ProwJobSpec struct {
 
 	// ReporterConfig holds reporter-specific configuration
 	ReporterConfig *ReporterConfig `json:"reporter_config,omitempty"`
+
+	// RerunPermissions holds information about which users can rerun the job
+	RerunPermissions *RerunPermissions `json:"rerun_permissions,omitempty"`
+}
+
+type RerunPermissions struct {
+	// If AllowAnyone is set to true, any user can rerun the job
+	AllowAnyone bool `json:"allow_anyone,omitempty"`
+	// Groups contains names of groups of users who can rerun the job
+	Groups []string `json:"groups,omitempty"`
+	// Users contains names of individual users who can rerun the job
+	Users []string `json:"users,omitempty"`
```

Comment:
`users` could also be made specific to GitHub users. I don't think the distinction is as important here, but it is probably still worth it so that we can add another identity provider in the future if we want.
@stevekuznetsov WDYT is that overkill?

Reply:
We're being specific for teams, might as well do it for users